src/manager/node.c

```diff
@@ -377,6 +378,55 @@ static int node_match_heartbeat(UNUSED sd_bus_message *m, UNUSED void *userdata,
         return 1;
 }
 
+static int node_on_match_proxy_new(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
+        Node *requesting_node = userdata;
+
+        const char *node_name = NULL;
+        const char *unit_name = NULL;
+        const char *mode = NULL;
+        const char *proxy_object_path = NULL;
+
+        int r = sd_bus_message_read(m, "ssso", &node_name, &unit_name, &mode, &proxy_object_path);
+        if (r < 0) {
+                hirte_log_errorf("Invalid arguments in ProxyNew signal: %s", strerror(-r));
+                return r;
+        }
+        hirte_log_debugf(
+                        "Matched ProxyNew signal from node '%s' for unit '%s' on node '%s'",
+                        requesting_node->name,
+                        unit_name,
+                        node_name);
+
+        Manager *manager = requesting_node->manager;
+
+        Node *providing_node = manager_find_node(manager, node_name);
+        if (providing_node == NULL) {
+                hirte_log_errorf("Could not find node with name '%s'", node_name);
+                return -ENXIO;
+        }
+
+        _cleanup_proxy_monitor_ ProxyMonitor *monitor = proxy_monitor_new(
+                        manager, requesting_node, proxy_object_path, providing_node);
+        if (monitor == NULL) {
+                hirte_log_error("Failed to create proxy monitor, OOM");
+                return -ENOMEM;
+        }
+        LIST_APPEND(monitors, manager->proxy_monitors, proxy_monitor_ref(monitor));
```

Comment:
I don't think we should add the monitor to the list untill we know everything is successful (i.e. the subscription proxy new call succeeds. Otherwise we'll get a half-working proxy in the list which will never complete.